Makefile

```diff
@@ -9,10 +9,8 @@ PROJ = google_containers
 
 PKG_SOURCES := $(shell find pkg -name '*.go')
 
-node-problem-detector: ./bin/node-problem-detector
-
-./bin/node-problem-detector: $(PKG_SOURCES) node_problem_detector.go
-	CGO_ENABLED=0 GOOS=linux godep go build -a -installsuffix cgo -ldflags '-w' -o node-problem-detector
+node-problem-detector: $(PKG_SOURCES) node_problem_detector.go
+	GOOS=linux go build -ldflags '-w -extldflags "-static"' -o node-problem-detector
```

Comment:
@andyxning There is still some problem here.
`go-systemd` is using `dlopen` to load `libsystemd-dev`. `-static` flag causes the following error:
```
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x0]

runtime stack:
runtime.throw(0x1a01fd9, 0x2a)
	/usr/local/go/src/runtime/panic.go:566 +0x95
runtime.sigpanic()
	/usr/local/go/src/runtime/sigpanic_unix.go:12 +0x2cc

goroutine 1 [syscall, locked to thread]:
runtime.cgocall(0x165b210, 0xc4204af8e0, 0xc400000000)
	/usr/local/go/src/runtime/cgocall.go:131 +0x110 fp=0xc4204af898 sp=0xc4204af858
k8s.io/node-problem-detector/vendor/github.com/coreos/pkg/dlopen._Cfunc_dlopen(0x7fdcfc000970, 0x1, 0x0)
	k8s.io/node-problem-detector/vendor/github.com/coreos/pkg/dlopen/_obj/_cgo_gotypes.go:95 +0x4e fp=0xc4204af8e0 sp=0xc4204af898
k8s.io/node-problem-detector/vendor/github.com/coreos/pkg/dlopen.GetHandle(0x2645e40, 0x4, 0x4, 0x0, 0x0, 0x0)
	/usr/local/google/home/lantaol/workspace/src/k8s.io/node-problem-detector/vendor/github.com/coreos/pkg/dlopen/dlopen.go:45 +0xd4 fp=0xc4204af948 sp=0xc4204af8e0
k8s.io/node-problem-detector/vendor/github.com/coreos/go-systemd/sdjournal.getFunction(0x19f3f88, 0x19, 0x0, 0x0, 0x0)
	/usr/local/google/home/lantaol/workspace/src/k8s.io/node-problem-detector/vendor/github.com/coreos/go-systemd/sdjournal/functions.go:46 +0x1d4 fp=0xc4204af9a8 sp=0xc4204af948
k8s.io/node-problem-detector/vendor/github.com/coreos/go-systemd/sdjournal.NewJournalFromDir(0xc420486600, 0xc, 0xc42040c890, 0x0, 0x0)
	/usr/local/google/home/lantaol/workspace/src/k8s.io/node-problem-detector/vendor/github.com/coreos/go-systemd/sdjournal/journal.go:377 +0xa4 fp=0xc4204afa58 sp=0xc4204af9a8
k8s.io/node-problem-detector/pkg/kernelmonitor/logwatchers/journald.getJournal(0xc4204865f0, 0x8, 0xc420486600, 0xc, 0xc4204865f8, 0x3, 0xc4204afc40, 0x465cf3, 0x268c1a0)
	/usr/local/google/home/lantaol/workspace/src/k8s.io/node-problem-detector/pkg/kernelmonitor/logwatchers/journald/log_watcher.go:133 +0x1d4 fp=0xc4204afbd8 sp=0xc4204afa58
k8s.io/node-problem-detector/pkg/kernelmonitor/logwatchers/journald.(*journaldWatcher).Watch(0xc420010460, 0x1, 0x1, 0x17b9f00)
	/usr/local/google/home/lantaol/workspace/src/k8s.io/node-problem-detector/pkg/kernelmonitor/logwatchers/journald/log_watcher.go:62 +0x47 fp=0xc4204afc50 sp=0xc4204afbd8
k8s.io/node-problem-detector/pkg/kernelmonitor.(*kernelMonitor).Start(0xc42000f260, 0x0, 0x1a86701, 0xb)
	/usr/local/google/home/lantaol/workspace/src/k8s.io/node-problem-detector/pkg/kernelmonitor/kernel_monitor.go:86 +0xb5 fp=0xc4204afcb8 sp=0xc4204afc50
k8s.io/node-problem-detector/pkg/problemdetector.(*problemDetector).Run(0xc4203ee1b0, 0xc42000f260, 0x26506a0)
	/usr/local/google/home/lantaol/workspace/src/k8s.io/node-problem-detector/pkg/problemdetector/problem_detector.go:56 +0x66 fp=0xc4204afec0 sp=0xc4204afcb8
main.main()
	/usr/local/google/home/lantaol/workspace/src/k8s.io/node-problem-detector/node_problem_detector.go:36 +0x6e fp=0xc4204aff08 sp=0xc4204afec0
runtime.main()
	/usr/local/go/src/runtime/proc.go:183 +0x1f4 fp=0xc4204aff60 sp=0xc4204aff08
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2086 +0x1 fp=0xc4204aff68 sp=0xc4204aff60

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2086 +0x1

goroutine 5 [chan receive]:
k8s.io/node-problem-detector/vendor/github.com/golang/glog.(*loggingT).flushDaemon(0x268c1a0)
	/usr/local/google/home/lantaol/workspace/src/k8s.io/node-problem-detector/vendor/github.com/golang/glog/glog.go:879 +0x7a
created by k8s.io/node-problem-detector/vendor/github.com/golang/glog.init.1
	/usr/local/google/home/lantaol/workspace/src/k8s.io/node-problem-detector/vendor/github.com/golang/glog/glog.go:410 +0x21d

goroutine 19 [chan receive]:
k8s.io/node-problem-detector/pkg/condition.(*conditionManager).syncLoop(0xc420011950)
	/usr/local/google/home/lantaol/workspace/src/k8s.io/node-problem-detector/pkg/condition/manager.go:94 +0x6e
created by k8s.io/node-problem-detector/pkg/condition.(*conditionManager).Start
	/usr/local/google/home/lantaol/workspace/src/k8s.io/node-problem-detector/pkg/condition/manager.go:79 +0x3f
```

There may be some way to fix this, I'll figure it out.
If it's hard to fix, we may have to:
* Disable CGO when not building journald support, so the binary will be statically linked automatically.
* Enable CGO when building journald support, so the binary will be dynamically linked.

For standalone NPD, either use non-journald version (statically linked) or use journald version (takes care of shared library).